cmd/snap-bootstrap/cmd_initramfs_mounts.go

```diff
@@ -282,43 +295,96 @@ func generateMountsModeRecover(mst *initramfsMountsState) error {
 	}
 
 	// 3. mount ubuntu-data for recovery
+	// start out assuming we have everything and lower expectations as things
+	// fail on us
+	haveHost := true
```

Comment:
maybe it should be named haveHostData or haveData and then the other `haveHostSave`?